app/controllers/mixins/breadcrumbs_mixin.rb

```diff
@@ -108,14 +119,10 @@ def current_tree_path(tree, active_node, path = [])
 
     def build_breadcrumbs_from_tree
       breadcrumbs = []
-      node = x_node
-      tree = build_tree if node
 
-      if tree.present?
-        tree.tree_nodes.each do |subtree|
-          value = current_tree_path(subtree, node)
-          breadcrumbs = value if value
-        end
+      @trees.find { |tree| tree.name == x_active_tree }.tree_nodes.each do |subtree|
+        value = current_tree_path(subtree, x_node)
```

Comment:
You don't need to go through the `@trees` here, just use `TreeBuilder.extract_node_model_and_id(x_node)` and then find the record. But I guess `@record` can be the same sometimes...

Reply:
This works perfectly for `record` nodes.. but what about folders and other nodes, which does not have "true" objects. Can I build their TreeNode if I know their ids?

Reply:
You mean `Hash` nodes? The ones I'm trying to get rid of? Well, the method should work on them, but you have no access to their title I guess. I will look into this a little more, until then use your current solution as a fallback if the model is `Hash`.

Reply:
Done!